src/utils/parser/scopes.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import toPairs from "lodash/toPairs";
+import uniq from "lodash/uniq";
+
+import type { Scope } from "./types";
+
+function getScopeVariables(scope: Scope) {
+  const { bindings } = scope;
+
+  return toPairs(bindings).map(([name, binding]) => ({
+    name,
+    references: binding.referencePaths
+  }));
+}
+
+function getScopeChain(scope: Scope): Scope[] {
+  const scopes = [scope];
+
+  do {
+    scopes.push(scope);
+  } while ((scope = scope.parent));
```

Comment:
so... i dropped the extra scope. don't know what to say about do/while. it terminates lol...